compiler/front/nimconf.nim

```diff
@@ -422,44 +422,36 @@ proc getSystemConfigPath*(conf: ConfigRef; filename: RelativeFile): AbsoluteFile
     if not fileExists(result): result = p / RelativeDir"etc/nim" / filename
     if not fileExists(result): result = AbsoluteDir"/etc/nim" / filename
 
-proc loadConfigs(
-    N: var NimConfParser, cfg: RelativeFile, cache: IdentCache
-  ) =
-  setDefaultLibpath(N.config)
-
-  proc readConfigFile(N: var NimConfParser, path: AbsoluteFile) =
-    let configPath = path
-    if readConfigFile(N, configPath, cache):
-      N.config.configFiles.add(configPath)
-
+iterator configFiles(N: NimConfParser, cfg: RelativeFile): AbsoluteFile =
```

Comment:
Only a very soft suggestion, but given that only the `N.config` is used, a `ConfigRef` could be passed directly. If this conflicts with your plans for the iterator (e.g. opening lexers), feel free to disregard the suggestion.

Reply:
I can do this after the merge if it makes sense. I'm trying to make all these little PRs close enough to the big one (`saem-cli-remove-legacy-reports`) so I can shrink the delta between it and `devel`.

I'm also not quite sure where things are going to land, because the `ConfigRef` we're passing around should probably be "just data" and no user visible actions be taken based on it. Then when we know it's good it should get merged into the "real" `ConfigRef`... but that approach might be off for reasons I don't quite understand as of yet.